pkg/service/auth.go

```diff
@@ -228,6 +230,9 @@ func (a *AuthService) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 // Check performs authorization check based on the attributes associated with the incoming request,
 // and returns status `OK` or not `OK`.
 func (a *AuthService) Check(parentContext gocontext.Context, req *envoy_auth.CheckRequest) (*envoy_auth.CheckResponse, error) {
+	_, span := otel.Tracer("AuthService").Start(parentContext, "Check")
```

Comment:
I was thinking maybe on adding attribute to the span with the request id, maybe even look into a [propagator](https://opentelemetry.io/docs/instrumentation/go/manual/#propagators-and-context) compatible with W3C Trace Context format (https://www.w3.org/TR/trace-context). WDYT?

Reply:
@guicassolato Yes, adding `request id` to the span metadata will be useful for filtering and sorting. After doing some research I found out that can be done by: 
- SpanAttributes: To add contextual information that is relevant to a specific span, but doesn't need to be propagated to downstream services.
ex : `span.SetAttributes(attribute.String(request))`

- Propagators (Our scenario): To pass contextual information between services in a distributed system. I assume this is how we should implement 
ex :   
```
propagator.Inject(ctx, propagation.HeaderCarrier(requestId)) 
propagator.Extract(r.Context(), propagation.HeaderCarrier(requestId))
 ```